nifi-nar-bundles/nifi-hive-bundle/nifi-hive-processors/src/main/java/org/apache/nifi/processors/hive/SelectHiveQL.java

```diff
@@ -310,6 +311,15 @@ private void onTrigger(final ProcessContext context, final ProcessSession sessio
         try (final Connection con = dbcpService.getConnection();
              final Statement st = (flowbased ? con.prepareStatement(selectQuery) : con.createStatement())
         ) {
+            try {
+                final int queryTimeout = context.getProperty(QUERY_TIMEOUT).evaluateAttributeExpressions(fileToProcess).asInteger();
```

Comment:
This looks fungible with the logic from `PutHive`. Why not move it to the abstract base class as a method both can reuse?

Reply:
Good point. I just pushed a commit to address it.